ocw/lib/azure.py

```diff
@@ -150,7 +150,7 @@ def cleanup_blob_containers(self) -> None:
                 self.log_dbg(f'Found container {container.name}')
                 container_blobs = self.container_client(container.name).list_blobs()
                 for blob in container_blobs:
-                    if self.is_outdated(blob.last_modified):
+                    if self.is_outdated(blob.last_modified) or self.is_cancelled(blob):
```

Comment:
I would not propagate this on disks , images and everything which goes into cleanup_all logic . this entities has different lifecycle and not tightly coupled to certain job . also this resource type NOT always have needed tags ( actually better to say in most cases they do NOT have them ) 

Also please check this [URL ](https://openqa.suse.de/tests/11560368#dependencies)  to see why it is not gonna work